pkg/registry/core/pod/strategy.go

```diff
@@ -303,11 +303,17 @@ func ToSelectableFields(pod *api.Pod) fields.Set {
 	// amount of allocations needed to create the fields.Set. If you add any
 	// field here or the number of object-meta related fields changes, this should
 	// be adjusted.
-	podSpecificFieldsSet := make(fields.Set, 9)
+	podSpecificFieldsSet := make(fields.Set, 10)
 	podSpecificFieldsSet["spec.nodeName"] = pod.Spec.NodeName
 	podSpecificFieldsSet["spec.restartPolicy"] = string(pod.Spec.RestartPolicy)
 	podSpecificFieldsSet["spec.schedulerName"] = string(pod.Spec.SchedulerName)
 	podSpecificFieldsSet["spec.serviceAccountName"] = string(pod.Spec.ServiceAccountName)
+	if pod.Spec.SecurityContext != nil {
+		podSpecificFieldsSet["spec.hostNetwork"] = strconv.FormatBool(pod.Spec.SecurityContext.HostNetwork)
```

Comment:
I fail to see the relation with the `SecurityContext` :/

Reply:
The `hostNetwork`  field is in the internal core.PodSpec.SecurityContext and corev1.PodSpec.HostNetwork

Reply:
yeah, but then what is the use case we are trying to solve?

is not
```
kubectl get pods --field-selector spec.hostNetwork=true"
```

Reply:
Yes, in my test env it did work.
```
[root@master0 yz]# kubectl get po -ojsonpath='{.items[*].spec.hostNetwork}'
true
[root@master0 yz]# kubectl get po
NAME   READY   STATUS    RESTARTS   AGE
yz-1   1/1     Running   0          3m47s
yz-2   1/1     Running   0          3m41s
[root@master0 yz]# oc get po --field-selector=spec.hostNetwork=true  
NAME   READY   STATUS    RESTARTS   AGE
yz-1   1/1     Running   0          3m57s
```

Reply:
Internal doesn't matter - fieldpaths are on the external (versioned) types only